MetaMask/metamask-extension
# Replacing setTimeout in auto-lock time limit with chrome alarm

#15929 
Replacing the setTimeout used to auto lock the application based on a time limit with chrome alarm.

Manual testing:
1. Log into MM and go to Advanced settings >> Auto-lock timer (minutes), and set a value for the auto lock.
2. Leave the app inactive for the above-mentioned time, and the app should lock the user out.


Points to verify:
1. The addListener should get triggered only once before logging out.
2. The alarm gets created every time user log in.
3. The timeout works even after the service worker restart
